enforcer.go

```diff
@@ -307,17 +307,21 @@ func (e *Enforcer) LoadPolicy() error {
 
 	if e.autoBuildRoleLinks {
 		needToRebuild = true
-		for _, rm := range e.rmMap {
-			err := rm.Clear()
+		newRmMap := make(map[string]rbac.RoleManager)
+		for ptype, rm := range e.rmMap {
+			newRmMap[ptype] = rm.Copy()
+			err := newRmMap[ptype].Clear()
```

Comment:
Create an empty ``rm`` directly (use ``defaultrolemanager.NewRoleManager(10)``), instead of copying and clearing

Reply:
> Create an empty `rm` directly (use `defaultrolemanager.NewRoleManager(10)`), instead of copying and clearing

We can't use `defaultrolemanager.NewRoleManager(10)` because `Enforcer` can have custom `RoleManager`. See here: https://github.com/casbin/casbin/blob/master/enforcer.go#L263

And my first implementation was with `defaultrolemanager.NewRoleManager(10)`, but tests failed. I think problem is that `defaultrolemanager.NewRoleManager(10)` losts some data in `RoleManager`. 
Anyway, copyign will be better with custom `RoleManager`s, i think.